### PR TITLE
refactor: convert Java to Kotlin

### PR DESCRIPTION
**Pull request by kacperkapusciak**
## Description
This PR converts all Java android-connected files to Kotlin. Also it adds ktlint and ktlintFormat as linter and formatter for Kotlin code.

## Changes

- Updated Gradle from 4.10 to 6.9
- Converted Java to Kotlin
- Set official Kotlin code style for formatting
- Added ktlint and ktlintFormat tasks per commit

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -39,4 +49,30 @@ dependencies {
...
+configurations {
+    ktlint
+}
+
+dependencies {
+    ktlint "com.pinterest:ktlint:0.40.0"
+}
+
+task ktlint(type: JavaExec, group: "verification") {
+    description = "Check Kotlin code style."
```
Comment (WoLewicki) [inline android/build.gradle:65]: Is it somehow possible to check if there are any kotlin files staged inside the script? For now it fails if you have unstaged changes in kotlin files. If it is not possible, then we can leave as it is now.

**Review by WoLewicki — COMMENTED** [review]
Added some comments to the code

File: package.json [hunk 2]
```diff
@@ -6,12 +6,14 @@
     "check-types": "tsc --noEmit",
     "start": "react-native start",
     "test:unit": "jest --passWithNoTests",
-    "format": "yarn format-js && yarn format-android && yarn format-ios && yarn format-windows",
     "format-js": "prettier --write --list-different './{src,Example}/**/*.{js,ts,tsx}'",
```
[the remark this answers is not in the document]
Comment (kacperkapusciak) [reply package.json:9]: this command is also run on the CI so I think it should be

File: .husky/pre-commit [hunk 3]
```diff
@@ -2,4 +2,5 @@
 . "$(dirname "$0")/_/husky.sh"
 
 yarn lint-staged
+yarn lint-android
 yarn check-types
```
[the remark this answers is not in the document]
Comment (kacperkapusciak) [reply .husky/pre-commit:6]: good catch! i've removed it in one of the commits but it came back as a post-merge issue. i'll remove it here but also add `yarn check-types` to the CI script

Fixed in e461d8dbf3ee08fd931d29df4100ae0c279b36a6 & f97e0e699bbe9baba1be1439d15ed2f87059474d

File: android/build.gradle (hunk 1 above)
Comment (kacperkapusciak) [reply android/build.gradle:65]: Fixed in 44e9db403908ac48f5a6379c65074676ebd416cb

**Review by WoLewicki — COMMENTED** [review]
Very good job!

**Review by WoLewicki — APPROVED** [review]

**Pull request merged**